### ISSUE TITLE
IDE crash on Ubuntu 17.10

### ISSUE THREAD

**Issue by nikobonomi** [jMonkeyEngine/sdk#160]
On Ubuntu 17.10 jmonkey  IDE crash on the upper menu (File, Edit, View, ....) and also when I use the right click.

I attach a log that I got ftom the JVM.

I tried with a fresh install off ubuntu and jmonkey and I got the same problem.

[hs_err_pid2762.log](https://github.com/jMonkeyEngine/sdk/files/1818375/hs_err_pid2762.log)

OS: Ubuntu 17.10
Laptop: Dell XPS 13 9343-9331
Jmonkey: v3.2.1-stable-sdk3

Sorry for my bed english :) I hope that is understandable

**Comment by grizeldi** [comment]
Didn't try the 3.2.1 stable sdk3 release yet, but stable sdk1 seems to be working just fine here (also ubuntu 17.10). It might be hardware related.

@nikobonomi paste the logs to gist, pastebin or something like that next time please, because as is I can't open it on mobile 😉

**Comment by MeFisto94** [comment]
A quick internet research showed this:
https://bugs.openjdk.java.net/browse/JDK-8170519
https://bugs.openjdk.java.net/browse/JDK-8172142

That means we can't do much since it's related to libawt / libX but it could be that this is just a matter of the jdk version. It seems to be related to the fact that 17.10 now ships wayland and the theme/awt cannot mess with that.

I don't understand it though since I am also using 17.10, but not a clean install, only dist-upgraded from 16.10 back then.

If you are bored feel free to try around and maybe also report your system java version (and or maybe if you used the installer it uses the bundled jdk where then old sdk1 could really help.

**Comment by grizeldi** [comment]
17.10 upgraded from any previous version doesn't have wayland unless specificaly installed. At least mine doesnt.

**Comment by nikobonomi** [comment]
Thanks guys for the quick replies.

I also tried the sdk1 stable but I got the same problem.

I use the JDK that come with JMonkey, but I tried also with my java and I got the same error.
My Java version is:  "1.8.0_161"

I managed to get it working in a strange way:
I must have 2 monitor, and Chrome closed. And JMonkey work only in one of the 2 monitors. It's quite strange.
If necessary, I can record a video.

I tried Jmonkey on 2 computer and get the same problem :/ 

Just for testing I tried with the java 9. And the issue was gone, but some GUI interfaces have disappears.

**Comment by MeFisto94** [comment]
@grizeldi But a fresh 17.10 Install comes with Wayland?

> If necessary, I can record a video.
I am afraid there is not much we can do about it. What might be possible is to try to use the default theme, which might not use that particular bugged feature (are you using netbeans by any chance?)
A way to achieve this might be to rename/delete the jme3-dark-laf.nbm however I can't tell what happens on a non-clean install, maybe the sdk crashes.

The clean way would be https://hub.jmonkeyengine.org/t/darkmonkey-as-the-new-sdk-default/35419 (see "I don't want Dark Monkey", however I guess you are unable to open this menus?)

> I tried with the java 9
That probably means that bug is fixed in java 9 but not in java 8, which would be a bummer, especially because Netbeans for Java 9 is currently in beta and I can't estimate how much work it is to adopt our code (because of java 9's versioning stuff)

**Comment by grizeldi** [comment]
Yes, clean 17.10 install comes with wayland as default and X as backup.

**Comment by nikobonomi** [comment]
@MeFisto94 Yes it comes with wayland. So, maybe with the 18.04 the problem disappear because they woll roll back to xorg? or at least this is what I understand.

No, i cant reach this menu.

**Comment by nikobonomi** [comment]
Hi all, switching to xorg is the solution.
The problem is running jmonkey on wayland.

Thanks @grizeldi and @MeFisto94 for the support.

**Comment by grizeldi** [comment]
It would probably still be good if we managed to fix this issue.

**Comment by grizeldi** [comment]
Since this is probably a JVM problem and latest Ubuntu rolled back to X display server, I think we can close this for the time being.
